src/store/hooks/useAuth.tsx

```diff
@@ -0,0 +1,223 @@
+import {
+  createContext, useContext, useMemo, ReactNode,
+  useEffect, useState,
+} from 'react';
+import {
+  getAuth, onAuthStateChanged, User, signOut, Auth,
+} from 'firebase/auth';
+import { LoadingOverlay } from '@mantine/core';
+import { useStorageEngine } from '../storageEngineHooks';
+import { FirebaseStorageEngine } from '../../storage/engines/FirebaseStorageEngine';
+import { GlobalConfig } from '../../parser/types';
+
+interface LocalStorageUser{
+  name: string,
+  email: string
+}
+
+interface UserWrapped{
+  user:User|LocalStorageUser|null,
+  determiningStatus:boolean,
+  isAdmin:boolean
+}
+
+type verifyAdminStatusInput = User | null | LocalStorageUser;
+
+// Defines default AuthContextValue
+interface AuthContextValue {
+  user: UserWrapped;
+  adminVerification:boolean;
+  login: (user: UserWrapped) => void;
+  logout: () => void;
```

Comment:
Since these modify state they're async

```suggestion
  login: (user: UserWrapped) => Promise<void>;
  logout: () => Promise<void>;
```